annotations/views/relationtemplate_views.py

```diff
@@ -201,15 +202,76 @@ def create_from_relationtemplate(request, template_id):
         text = get_object_or_404(Text, pk=data['occursIn'])
         project_id = data.get('project')
         if project_id is None:
-            project_id = VogonUserDefaultProject.objects.get(for_user=request.user).project.id
-        relationset = relations.create_relationset(template, data, request.user, text, project_id)
+            project_id = VogonUserDefaultProject.objects.get(
+                for_user=request.user).project.id
+        relationset = relations.create_relationset(
+            template, data, request.user, text, project_id)
         response_data = {'relationset': relationset.id}
-    else:   # Not sure if we want to do anything for GET requests at this point.
+    else:  # Not sure if we want to do anything for GET requests at this point.
         response_data = {}
 
     return JsonResponse(response_data)
 
 
+def create_from_text(request, template_id):
+    if request.method == 'POST':
+        data = json.loads(request.body)
+        appellations = data['appellations']
+        text_appellation = data['textAppellation']
+        template = get_object_or_404(RelationTemplate, pk=template_id)
+        text = get_object_or_404(Text, pk=data['occursIn'])
+        project_id = data.get('project')
+        if project_id is None:
```

Comment:
```if not project_id```? or could project_id be empty string and that would be valid?

Reply:
I did not write this part of the method and figured if it aint broke don't fix so I can not say for sure. From what I can tell I do think that it is possible for it to be a string in some cases. When the project id is initially set in the vue js data properties it is set as a string so if for some reason it did not have a value I am assuming that it would set an empty which then could be submitted elsewhere.